plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -751,3 +751,61 @@ spec:
           protocol: TCP
 `)
 }
+
+func TestPatchTransformerMergeKeyList(t *testing.T) {
```

Comment:
this is core functionality.  

we should  record the outcome in test form for several cases, and if need be debate/change that outcome.

some cases 

 -  incoming deployment has bad data, e.g. two `ports` entries with matching `{containerPort,protocol}` entries but different names.  We add a label via a _patch_ (not via the label transformer) to trigger patch/merge code.  One of the ports entries should disappear since no two entries can have the same merge key.  This test records the resolution of #2767 (even though it doesn't allow name aliases for ports). 

 -  a patch to a deployment adds a port with a name to an existing ports list.  the new entry matches an existing but nameless `{containerPort,protocol}`.  this merely adds a name to the port, should not add a new port.

 -  same as before, but the original port has a name, and we want to change that name.

 - same as above, but no protocol.

 - same as above, but no container port. that should be an error.  

maybe more combinations?

Reply:
Also, a better place to keep these tests is

https://github.com/kubernetes-sigs/kustomize/blob/master/api/filters/patchstrategicmerge/patchstrategicmerge_test.go

and/or

https://github.com/kubernetes-sigs/kustomize/blob/master/api/filters/patchstrategicmerge/example_test.go